modules/error_handling/include/hephaestus/error_handling/panic.h

```diff
@@ -66,15 +74,19 @@ class Panic : public std::runtime_error {
 /// @param message A message describing the error and what caused it
 /// @param location Location in the source where the error was triggered at
 template <typename... Args>
-void panic(internal::StringLiteralWithLocation<Args...> message, Args&&... args) {
+void panic(error_handling::detail::StringLiteralWithLocation<Args...> message, Args&&... args) {
   std::string formatted_message = fmt::format(message.value, std::forward<Args>(args)...);
-#ifndef DISABLE_EXCEPTIONS
-  throw Panic{ std::move(formatted_message), message.location };
-#else
-  auto e = Panic{ std::move(formatted_message), message.location };
-  CHECK(false) << fmt::format("[ERROR {}] at {}:{}", e.what(), message.location.file_name(),
-                              message.location.line());
-#endif
+  auto location = std::string(utils::string::truncate(message.location.file_name(), "modules")) + ":" +
+                  std::to_string(message.location.line());
+  log(FATAL, "Program terminated with error", "error", std::move(formatted_message), "location",
+      std::move(location));
+  telemetry::flushLogEntries();
+
+  if (error_handling::panicAsException()) {
+    throw error_handling::PanicException(formatted_message);
+  }
```

Comment:
awesome to map this via scope! not sure if we need it though...
would be cleaner to have only 1 way to panic... what are use cases? zenoh?